i18n of Basic Campaign token type, property and light names

bubblobill opened the pull request:
### Identify the Bug or Feature request
progresses #4600 
https://github.com/RPTools/maptool/issues/4600#issuecomment-3016871037

### Description of the Change
i18n of

-  Basic token type,
-  property names
-  light names
-  sight names
-  state names
-  bar name

### Possible Drawbacks
Might not have caught all the method calls.

### Documentation Notes
Localisation of default Campaign setup language.

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/RPTools/maptool/5656)
<!-- Reviewable:end -->

bubblobill committed:
i18n of Basic Campaign token type and property strings.
Added i18nKey field to LightSource.
Generated new default lights xml

bubblobill committed:
spotpain

bubblobill committed:
missed this

bubblobill committed:
Added states, bars, sight

bubblobill commented:
Tested this with a garbage string `©§¦¤µ ¿ÃÒÑÐýø÷` in various i18n strings and it worked.
As far as I can tell, changing the name does no harm to any of the internationalised features.

kwvanderlinde commented on src/main/java/net/rptools/maptool/model/LightSource.java:
```diff
@@ -315,6 +380,7 @@ public int compareTo(@Nonnull LightSource o) {
   public static @Nonnull LightSource fromDto(@Nonnull LightSourceDto dto) {
     return new LightSource(
         dto.hasName() ? dto.getName().getValue() : null,
+        dto.hasI18NKey() ? dto.getI18NKey().getValue() : null,
```
Unless I'm missing something, we don't need the i18nKey in DTO since it's only real use is when reading / creating the default light sources.

kwvanderlinde commented:
FYI, I've added #5665 to move the default light sources into code. Whenever we do that, it will also remove the need for the I18N support in `LightSource` itself as we can properly customize the creation of these lights.

kwvanderlinde commented on src/main/java/net/rptools/maptool/model/LightSource.java:
```diff
@@ -283,7 +290,65 @@ public record LightArea(Light light, Area area) {}
+        listOut.add(
+            new LightSource(
+                nameOut,
+                ls.getI18nKey(),
```
One thing to watch out for - even though we need to read i18nKey from XML because of how default light sources are loaded, we shouldn't write it out into campaign files. I would suggest passing `null` here so the I18N key isn't carried around after loading.

kwvanderlinde commented:
Sorry for the merge conflicts...

Now that light sources are defined in code, the translation can be done in `CampaignProperties` when defining each light and `LightSource` no longer needs to support i18n directly. E.g.,
```
categorizedLights.addAllToCategory(
        "D20",
        List.of(
            createD20LightSource(I!8N.getString("Default.campaign.lightSource.light.candle"), 5),
…
```

bubblobill committed:
hopefully conflicts resolved and sanity restored

bubblobill committed:
spittle

bubblobill committed:
spittle

bubblobill committed:
spittle

kwvanderlinde commented on src/main/java/net/rptools/maptool/model/CampaignProperties.java:
Need to add the light source translations here.

bubblobill replied:
done

kwvanderlinde commented on src/main/resources/net/rptools/maptool/model/defaultLightSourcesMap.xml:
This file should be removed again as it is no longer used.

kwvanderlinde commented on src/main/proto/data_transfer_objects.proto:
```diff
@@ -73,6 +73,7 @@ message LightSourceDto {
   }
   repeated LightDto lights = 1;
   google.protobuf.StringValue name = 2;
+  google.protobuf.StringValue i18nKey = 7;
```
No more need for adding i18n keys to DTO.

bubblobill replied:
done

kwvanderlinde commented on src/main/java/net/rptools/maptool/model/LightSource.java:
```diff
@@ -296,7 +296,7 @@ public int compareTo(@Nonnull LightSource o) {
       if (nameLong != Integer.MIN_VALUE && onameLong != Integer.MIN_VALUE)
         return nameLong - onameLong;
-      return name.compareTo(o.name);
+      return Objects.requireNonNull(name).compareTo(Objects.requireNonNull(o.name));
```
These `requireNonNull()` calls don't actually do anything and could be removed. With or without them, a NullPointerException will be thrown.

bubblobill replied:
But IntelliJ tells me to...

bubblobill replied:
changed

bubblobill committed:
Added translation light keys.
Removed i18n from dto
Removed null checks in LightSource
removed redundant light-source xml

bubblobill committed:
Added translation light keys.
Removed i18n from dto
Removed null checks in LightSource
removed redundant light-source xml

bubblobill committed:
Merge branch 'RPTools:develop' into Text-Maintenance

bubblobill committed:
spotted
